### PR TITLE
Story/ccls 2003 org search

### PR DESCRIPTION
**Pull request by porritta**
Includes 
CCLS-2003 - Organisation Search
CCLS-2004 - Organisation Search Results
CCLS-2007 - Organisation Confirm
Also Too Many/No Results screens, which we didn't have stories for.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -72,9 +72,8 @@ repositories {
 }
 
 dependencies {
-
+	implementation 'uk.gov.laa.ccms.data:data-api:0.0.6-690e561-SNAPSHOT'
```
Comment (PhilDigitalJustice) [inline build.gradle:75]: have the data api changes been merged?

File: src/main/java/uk/gov/laa/ccms/caab/constants/SessionConstants.java [hunk 2]
```diff
@@ -112,6 +112,23 @@ public class SessionConstants {
    */
   public static final String APPLICATION_CLIENT_NAMES = "applicationClientNames";
 
+  /**
+   * Session attribute used to keep track of criteria when searching for an organisation.
+   */
+  public static final String ORGANISATION_SEARCH_CRITERIA = "orgSearchCriteria";
```
Comment (PhilDigitalJustice) [inline src/main/java/uk/gov/laa/ccms/caab/constants/SessionConstants.java:118]: do we want full names here - organisationSearchCriteria

File: src/main/java/uk/gov/laa/ccms/caab/controller/application/summary/OpponentsSectionController.java [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+  /**
+   * Handles the GET request to fetch and display the opponents and other parties for a specific
+   * application.
+   *
+   * @param applicationId The id of the application, retrieved from the session.
+   * @param model The Model object to add attributes to for the view.
+   * @param session The HttpSession object to add attributes to for the session.
+   * @return The name of the view to be rendered.
+   */
+  @GetMapping("/application/summary/opponents")
```
Comment (PhilDigitalJustice) [inline src/main/java/uk/gov/laa/ccms/caab/controller/application/summary/OpponentsSectionController.java:85]: Should this be application/opponents/summary ?

File: src/main/resources/templates/application/opponents/opponents-organisation-shared-confirm.html [hunk 4]
```diff
@@ -0,0 +1,61 @@
+<!DOCTYPE html>
+<html lang="en" th:replace="~{layout :: layout(title=~{::title},mainContent=~{::#main-content},pageCategory=${'cases'},breadcrumbs=~{::#breadcrumbs})}" xmlns:th="http://www.thymeleaf.org">
+<head>
+    <title>LAA CCMS - Search for an organisation</title>
+</head>
+<body>
+<div class="govuk-width-container">
+    <div id="breadcrumbs">
+        <a href="/application/summary/opponents" class="govuk-back-link">Cancel and return to Opponents and other parties</a>
```
Comment (PhilDigitalJustice) [inline src/main/resources/templates/application/opponents/opponents-organisation-shared-confirm.html:9]: lowercase "opponents"

File: src/main/java/uk/gov/laa/ccms/caab/controller/application/summary/OpponentsSectionController.java (hunk 3 above)
Comment (porritta) [reply src/main/java/uk/gov/laa/ccms/caab/controller/application/summary/OpponentsSectionController.java:85]: I thought we were keeping a summary folder for the current snapshot of each application section. But can be moved if we're not doing that?

**Review by PhilDigitalJustice — COMMENTED** [review]
a few comments

File: src/main/java/uk/gov/laa/ccms/caab/controller/application/summary/OpponentsSectionController.java (hunk 3 above)
Comment (PhilDigitalJustice) [reply src/main/java/uk/gov/laa/ccms/caab/controller/application/summary/OpponentsSectionController.java:85]: looking at mine i chage proceeding and costs to:

@GetMapping("/application/proceedings-and-costs")

no summary at all 

But for proceedings directly i do:

@GetMapping("/application/proceedings/{proceeding-id}/summary")

File: src/main/java/uk/gov/laa/ccms/caab/constants/SessionConstants.java (hunk 2 above)
Comment (porritta) [reply src/main/java/uk/gov/laa/ccms/caab/constants/SessionConstants.java]: amended

File: src/main/java/uk/gov/laa/ccms/caab/constants/SessionConstants.java [hunk 5]
```diff
@@ -112,6 +112,23 @@ public class SessionConstants {
...
+
+  /**
+   * Session attribute used to keep track of organisation search results during the creation
+   * of a new application opponent.
+   * Used when returning to organisation search results screen to prepopulate table of results
+   */
+  public static final String ORGANISATION_SEARCH_RESULTS = "orgResultsPage";
```
[the remark this answers is not in the document]
Comment (porritta) [reply src/main/java/uk/gov/laa/ccms/caab/constants/SessionConstants.java]: amended

File: src/main/resources/templates/application/opponents/opponents-organisation-shared-confirm.html (hunk 4 above)
Comment (porritta) [reply src/main/resources/templates/application/opponents/opponents-organisation-shared-confirm.html]: done

File: build.gradle (hunk 1 above)
Comment (porritta) [reply build.gradle]: no idea why I was on that snapshot version. I've rolled it back to 0.0.5.

**Review by PhilDigitalJustice — APPROVED** [review]
LGTM

**Pull request merged**